pkg/ipamd/datastore/data_store_test.go

```diff
@@ -96,6 +96,57 @@ func TestDeleteENI(t *testing.T) {
 	assert.Error(t, err)
 	err = ds.RemoveENIFromDataStore("eni-1", true)
 	assert.NoError(t, err)
+
+}
+
+func TestDeleteENIwithPDEnabled(t *testing.T) {
+	ds := NewDataStore(Testlog, NullCheckpoint{}, true)
+
+	err := ds.AddENI("eni-1", 1, true, false, false)
+	assert.NoError(t, err)
+
+	err = ds.AddENI("eni-2", 2, false, false, false)
+	assert.NoError(t, err)
+
+	err = ds.AddENI("eni-3", 3, false, false, false)
+	assert.NoError(t, err)
+
+	err = ds.AddENI("eni-4", 4, false, false, false)
+	assert.NoError(t, err)
+
+	eniInfos := ds.GetENIInfos()
+	assert.Equal(t, len(eniInfos.ENIs), 4)
+
+	err = ds.RemoveENIFromDataStore("eni-2", false)
+	assert.NoError(t, err)
+
+	eniInfos = ds.GetENIInfos()
+	assert.Equal(t, len(eniInfos.ENIs), 3)
+
+	err = ds.RemoveENIFromDataStore("unknown-eni", false)
+	assert.Error(t, err)
+
+	eniInfos = ds.GetENIInfos()
+	assert.Equal(t, len(eniInfos.ENIs), 3)
+
+	// Add a prefix and assign a pod
+	ipv4Addr := net.IPNet{IP: net.ParseIP("10.0.0.0"), Mask: net.IPv4Mask(255, 255, 255, 240)}
+	err = ds.AddIPv4CidrToStore("eni-4", ipv4Addr, true)
+	assert.NoError(t, err)
+	ip, device, err := ds.AssignPodIPv4Address(
+		IPAMKey{"net1", "sandbox1", "eth0"},
+		IPAMMetadata{K8SPodNamespace: "default", K8SPodName: "sample-pod"})
+	assert.NoError(t, err)
+	assert.Equal(t, "10.0.0.0", ip)
+	assert.Equal(t, 4, device)
+
+	// Test force removal.  The first call fails because eni-1 has an IP with a pod assigned to it,
+	// but the second call force-removes it and succeeds.
+	err = ds.RemoveENIFromDataStore("eni-4", false)
+	assert.Error(t, err)
+	err = ds.RemoveENIFromDataStore("eni-4", true)
+	assert.NoError(t, err)
+
```

Comment:
I think it is worth checking datastore count here, then draining all ENIs and verifying count goes to 0.

Reply:
Should this test also be checking `ds.Total`?

Reply:
This test case is to check force removal of ENIs. The total count is checked in the next test case.